prov/gni/src/gnix_cm_nic.c

```diff
@@ -190,24 +190,28 @@ static int __process_datagram(struct gnix_datagram *dgram,
 /*******************************************************************************
  * Internal API functions
  ******************************************************************************/
-/*
- * generate a cdm_id, use the 16 LSB of base_id from domain
- * with 16 MSBs being obtained from atomic increment of
- * a local variable.
- */
 
-int _gnix_get_new_cdm_id(struct gnix_fid_domain *domain, uint32_t *id)
+int _gnix_cm_nic_create_cdm_id(struct gnix_fid_domain *domain, uint32_t seed,
+			       uint32_t *id)
 {
 	uint32_t cdm_id;
-	int v;
 
-	v = atomic_inc(&gnix_id_counter);
 	cdm_id = (domain->cdm_id_seed & 0x0000FFFF) |
-			((uint32_t)v << 16);
+			((uint32_t)seed << 16);
```

Comment:
see above.   Note we could someday move to using something other than the getpid method for generating ```domain->cdm_id_seed```.  A more robust way might be to create a file in /var/tmp with a unique name based on the ```cookie``` (this is the cookie value supplied to ```GNI_CdmCreate```).  The processes which will be using this ```cookie``` could then take turns locking this file, reading the current value of a counter in this file, update the counter with a new file, then unlocking the file.